### PR TITLE
chore: add convolutional autoencoder

### PR DESCRIPTION
**Pull request by ChadElRicardo**

### CODE REVIEW

**Review by AFAgarap — COMMENTED** [review]
@ChadElRicardo Kindly share some sample generated images using the model in this module. Thank you

**Comment by ChadElRicardo** [comment]
> @ChadElRicardo Kindly share some sample generated images using the model in this module. Thank you
Hi @AFAgarap, sharing sample generated images of the convolutional model.

**Original images** on the _first row_ and **reconstructed images** on the _second row_. Thank you.

cae_fashionmnist

cae_mnist

**Comment by AFAgarap** [comment]
> > @ChadElRicardo Kindly share some sample generated images using the model in this module. Thank you
> > Hi @AFAgarap, sharing sample generated images of the convolutional model.
> 
> **Original images** on the _first row_ and **reconstructed images** on the _second row_. Thank you.
> 
> cae_fashionmnist
> 
> cae_mnist

May I know the experiment parameters? Like the optimization algorithm used, the size of mini-batches, learning rate, etc.

**Comment by ChadElRicardo** [comment]
> > > @ChadElRicardo Kindly share some sample generated images using the model in this module. Thank you
> > > Hi @AFAgarap, sharing sample generated images of the convolutional model.
> > 
> > 
> > **Original images** on the _first row_ and **reconstructed images** on the _second row_. Thank you.
> > cae_fashionmnist
> > cae_mnist
> 
> May I know the experiment parameters? Like the optimization algorithm used, the size of mini-batches, learning rate, etc.

I used **Adam optimizer** with a **learning rate of 1e-3** on **mini-batches of 128 for the training dataset**.
